andela/ah-the-jedi-backend
# #165305280 Users should report articles

### What does this PR do?
Adds reporting functionality to the app

### Description of Task(s) to be completed?
-An authenticated user should be able to report articles that violate terms of agreement or are plagiarized to the authors haven admin for removal and or any other actions as per the terms of engagement

### How should this be manually tested?

Clone this repository on local machine and CD into project folder
`$ git clone git@github.com:andela/ah-the-jedi-backend.git`
`$ cd ah-the-jedi-backend`

Check out to the PR's branch
`$ git checkout ft-reports-165305280`

Create and activate virtual environment
`$ python3 -m venv venv`
`$ source .venv/bin/activate`

Install app dependencies from requirements.txt
`$ pip install -r api/requirements.txt`

Make migrations and run the server
`$ python manage.py makemigrations`
`$ python manage.py migrate`
`$ python manage.py runserver`

**Use postman to test the endpoints**


           POST <your-localhost>/api/users/

- [x] **Create a super user using the terminal**

       python manage.py createsuperuser



       POST <your-localhost>/api/login/ 

       Use the token provided after login for authorization 


- [x] **Create an article**

      POST  <you-local-host>/api/articles/


- [x] **Report the article**

      POST  <you-local-host>/api/reports/

### Screenshot
<img width="943" alt="Screenshot 2019-05-15 at 22 56 49" src="https://user-images.githubusercontent.com/45200891/57805045-c3632380-7764-11e9-82ba-4ede31398f22.png">

- [x] **Get all reports of logged in user**

      GET  <you-local-host>/api/article/reports/

### Screenshot
<img width="941" alt="Screenshot 2019-05-15 at 23 47 55" src="https://user-images.githubusercontent.com/45200891/57808426-feb52080-776b-11e9-905e-dbd4b24b95e1.png">


- [x] **Get one report**

      GET  <you-local-host>/api/article/reports/<report-id>/

### Screenshot
<img width="941" alt="Screenshot 2019-05-15 at 22 57 31" src="https://user-images.githubusercontent.com/45200891/57805080-daa21100-7764-11e9-8af1-9f82c637c043.png">



- [x] **Update a report**

      PUT  <you-local-host>/api/reports/<report-id>/

### Screenshot
<img width="943" alt="Screenshot 2019-05-15 at 22 58 28" src="https://user-images.githubusercontent.com/45200891/57805147-fdccc080-7764-11e9-9e8a-1177161b7ce6.png">



- [x] **Delete a report**

      DELETE  <you-local-host>/api/reports/<report-id>/

### Screenshot
<img width="942" alt="Screenshot 2019-05-15 at 22 59 08" src="https://user-images.githubusercontent.com/45200891/57805176-12a95400-7765-11e9-80be-b7cba19e1bd4.png">



- [x] **Get all reports as admin**

      1. Sign in the super user
      2. Use the token to access this endpoint
      GET  <you-local-host>/api/report/get_all/

### Screenshot
<img width="941" alt="Screenshot 2019-05-15 at 23 00 13" src="https://user-images.githubusercontent.com/45200891/57805255-3b314e00-7765-11e9-91d7-098955b64008.png">


- [x] **Delete report as admin**

      1. Sign in the super user
      2. Use the token to access this endpoint
      DELETE  <you-local-host>/api/report/delete/<report-id>/

### Screenshot
<img width="942" alt="Screenshot 2019-05-15 at 23 01 42" src="https://user-images.githubusercontent.com/45200891/57805374-6d42b000-7765-11e9-9031-a251cba63635.png">


- [x] **Delete reported article as admin**

      1. Sign in the super user
      2. Use the token to access this endpoint
      DELETE  <you-local-host>/api/articles/delete/<article-slug>/

### Screenshot
<img width="941" alt="Screenshot 2019-05-15 at 23 37 44" src="https://user-images.githubusercontent.com/45200891/57807797-944fb080-776a-11e9-944e-230256386a6d.png">





### What are the relevant PT stories?
[#165305280](https://www.pivotaltracker.com/story/show/165305280)
